### PR TITLE
Add long options for bits and bytes

### PR DESCRIPTION
**Pull request by Hofer-Julian**
As discussed in https://github.com/nushell/nushell/pull/10597#issuecomment-1745692687

### CODE REVIEW

**Review by amtoine — COMMENTED** [review]
i personally like long options :ok_hand:

File: crates/nu-cmd-extra/src/extra/bits/shift_left.rs [hunk 1]
```diff
@@ -90,12 +90,12 @@ impl Command for BitsShl {
             },
             Example {
                 description: "Shift left a number with 1 byte by 7 bits",
-                example: "2 | bits shl 7 -n '1'",
+                example: "2 | bits shl 7 --number-bytes '1'",
```
Comment (jntrnr) [inline crates/nu-cmd-extra/src/extra/bits/shift_left.rs:93]: not related to this PR, but I wonder why `--number-bytes` takes a string. Looks odd.

**Comment by jntrnr** [comment]
@Hofer-Julian - yeah exactly, these examples are way more readable now.

File: crates/nu-cmd-extra/src/extra/bits/shift_left.rs (hunk 1 above)
Comment (Hofer-Julian) [reply crates/nu-cmd-extra/src/extra/bits/shift_left.rs:93]: Seems like this command was introduced by @hustcer with https://github.com/nushell/nushell/pull/6202.
Maybe they remember if there was a reason for this or if this is an oversight :)

Comment (hustcer) [reply crates/nu-cmd-extra/src/extra/bits/shift_left.rs:93]: > not related to this PR, but I wonder why `--number-bytes` takes a string. Looks odd.

Actually `--number-bytes` takes a int from the beginning, but changed to a string in 90b65018b

Comment (Hofer-Julian) [reply crates/nu-cmd-extra/src/extra/bits/shift_left.rs:93]: I might be confused here, but doesn't this mean it took a string from the very beginning?
https://github.com/hustcer/nushell/blob/45897db03fa4d5e0956e4b0038015a61d3299df1/crates/nu-command/src/bits/shift_left.rs#L29

Comment (hustcer) [reply crates/nu-cmd-extra/src/extra/bits/shift_left.rs:93]: Oh, I see, "it can be 1, 2, 4, 8, auto", `auto` is a string

Comment (Hofer-Julian) [reply crates/nu-cmd-extra/src/extra/bits/shift_left.rs:93]: Ah yeah :)

**Comment by amtoine** [comment]
it appears we are quite happy with these changes, let's go with them and see how it becomes clearer :pray:

**Pull request merged**